### PR TITLE
feat(web-scripts): Upgrade to ESLint ^6.8.0

### LINKED ISSUE #94: Upgrade to ESLint 6

### PR DESCRIPTION
**Pull request by josephmcasey**
Resolves #94

## Status

Work In Progress

## Feature Intent

Updates the developer and peer dependencies for ESLint by 1 major version.

## Testing Information

0. Pull down this PR
1. `cd web-scripts && git clean -xdf`
2. `nvm install 12.14.0`
3. `yarn lerna run bootstrap`
4. `yarn test` @ project root
5. `yarn lint` @ project root

## References

[`ESLint 6 Migration`](https://eslint.org/docs/user-guide/migrating-to-6.0.0)

## **BREAKING CHANGE:**

**Users of web-scripts that rely on ESLint 5 will see unexpected linting errors.**

### CODE REVIEW

**Comment by josephmcasey** [comment]
Status: I wasn't able to reproduce locally the CI/CD Pipeline's failure for node 12.x. At this time, I am hoping to better understand why the integration.test.ts file failed.

File: packages/eslint-config-typescript/package.json [hunk 1]
```diff
@@ -16,12 +16,12 @@
...
     "@typescript-eslint/parser": ">=2",
-    "eslint": ">=5.x"
+    "eslint": ">=6.x"
   },
   "publishConfig": {
     "access": "public"
   },
   "engines": {
-    "node": ">=10"
+    "node": ">=10.13.0"
```
Comment (josephmcasey) [inline packages/eslint-config-typescript/package.json]: > * [Node.js 10 (10.13.0 and above)](https://eslint.org/docs/user-guide/migrating-to-6.0.0#drop-node-6)

**Comment by josephmcasey** [comment]
Status: I've noticed that @fastfrwrd has resolved the root cause for this failure in #93 . As he stated

> fresh projects using web-scripts would need to set their own resolutions to get lodash in the right place to get postinstall to pass.

This is the reason that [integration.test.ts](https://github.com/spotify/web-scripts/blob/master/packages/web-scripts/src/integration.test.ts) fails due to the before step where the repo is setup on [line 161](https://github.com/spotify/web-scripts/blob/master/packages/web-scripts/src/integration.test.ts#L161).

**Comment by fastfrwrd** [comment]
Second verse, same as the first! I've made some updates to the integration test to get it working, and I rebased against #93. To avoid back and forth, I'll close this PR and open a fresh one which forks off it.

Thanks so much for these PRs @josephmcasey!

**Pull request closed without merging**